### DISCUSSION TITLE
Green 'https' on some sites

### DISCUSSION THREAD

**Discussion by xvybihal** [bromite/bromite#1445] [Q&A]
Hey, I am little confused. Some sites have 'https' in the address bar written in green, and some don't. 

I can't seem to get around, how this is decided. Compared two sites, I both manage. On both, bromite uses TLS 1.3, AES_128_GCM, X25519. But one has green https, and the other does not. None of them have mixed content.

What is the main factor, that decies, if the protocol will be greenified?

Thanks

**Comment by csagan5** [comment]
This is a Chromium question, see https://blog.chromium.org/2021/07/increasing-https-adoption.html; the lock will eventually be gone altogether.

Bromite does not have any special treatment for EV certificates so not sure why you see it sometimes in green and sometimes not.

**Reply by xvybihal** [reply]
> This is a Chromium question, see https://blog.chromium.org/2021/07/increasing-https-adoption.html; the lock will eventually be gone altogether. 

This is not padlock question.

> Bromite does not have any special treatment for EV certificates so not sure why you see it sometimes in green and sometimes not.

This is not about EV certs. 


I think i have found out conditions when the https is green and when it's not. It's when the the URI has a path. Not sure why, what does it want to indicate to user.

![01](https://user-images.githubusercontent.com/3757613/136542712-364ec17e-2a90-4b49-b2d6-12b736a3d644.png)
![02](https://user-images.githubusercontent.com/3757613/136542719-2749f0fa-8550-4eaa-929a-1873deb6c34e.png)

**Comment by uazo** [comment]
I think it may be because of what was said [here](https://github.com/bromite/bromite/pull/731#issuecomment-868291024)

**Reply by csagan5** [reply]
It does not seem related to the template URL; I would look into Omnibox rendering code.

**Reply by csagan5** [reply]
It is in fact here: https://github.com/chromium/chromium/blob/106.0.5249.72/chrome/browser/ui/views/location_bar/location_icon_view.cc#L64

The non-green color is probably due to legacy TLS versions being used.

**Reply by xvybihal** [reply]
> The non-green color is probably due to legacy TLS versions being used.

No, it is not (in my example).

**Reply by csagan5** [reply]
Then it is not using green because the security state is considered neutral: https://github.com/chromium/chromium/blob/106.0.5249.72/components/security_state/core/security_state.h#L40

I cannot tell why HTTPS URLs would be considered neutral, there is not much explanation there and it would require more diving into the code, it is probably some bit of logic in [here](https://github.com/chromium/chromium/blob/106.0.5249.72/components/security_state/core/security_state.cc).

**Comment by AE-boop** [comment]
Seni surtuk

**Reply by AE-boop** [reply]
what's it to you. look at your job from green

**Comment by uazo** [comment]
with next version (v107) the problem disappears, so I guess it could be an upstream bug
